src/components/ts_form/ts_input.tsx

```diff
@@ -0,0 +1,43 @@
+import type { InputGroupProps } from '@blueprintjs/core';
+import { Classes, FormGroup, InputGroup } from '@blueprintjs/core';
+import styled from '@emotion/styled';
+import type { ChangeEvent } from 'react';
+import { useCallback, useMemo } from 'react';
+
+import { useFieldContext } from './context/use_ts_form.js';
+
+type TSInputProps = Omit<InputGroupProps, 'defaultValue' | 'name'>;
+
+const Input = styled(InputGroup)<{ error: boolean }>`
+  .${Classes.INPUT} {
+    border: 1px solid ${(props) => (props.error ? 'red' : 'inherit')};
```

Comment:
You should use Blueprint's intent properties to style the input and avoid hardcoding colors. If you absolutely need to use a color like this, then you should import the color from Blueprint. But it is probably not needed.